pkg/gke/util/gcp.ts

```diff
@@ -0,0 +1,159 @@
+ * @param projectId gcp project in which to make the request
+ * @param location object containing either gcp region or zone in which to make the request - defaults to DEFAULT_GCP_ZONE if both zone and region are undefined. region preempts zone
+ * @returns
+ */
+function getGKEOptions(resource: string, store: any, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}, clusterId?:string ) {
+  if (!cloudCredentialId || !projectId) {
+    return new Promise((resolve, reject) => reject(new Error('cloud credential or project id missing')));
+  }
+  if (!location.zone && !location.region) {
+    location.zone = DEFAULT_GCP_ZONE;
+  }
+  let params: QueryParams = {};
+
+  if (location.region) {
+    params = {
+      cloudCredentialId, projectId, region: location.region
+    };
+  } else if (location.zone) {
+    params = {
+      cloudCredentialId, projectId, zone: location.zone
+
+    };
+  }
+  if (clusterId) {
+    params.clusterId = clusterId;
+  }
+
+  const url = addParams(`/meta/${ resource }`, params);
+
+  return store.dispatch('management/request', {
+    url,
+    method:               'POST',
+    redirectUnauthorized: false,
+  });
+}
+
+export function getGKEZones(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKEZonesResponse> {
+  return getGKEOptions('gkeZones', store, cloudCredentialId, projectId, location);
+}
+
+export async function getGKEVersions(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKEVersionsResponse> {
+  return await getGKEOptions('gkeVersions', store, cloudCredentialId, projectId, location);
+}
+
+export function getGKEMachineTypes(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKEMachineTypesResponse> {
+  return getGKEOptions('gkeMachineTypes', store, cloudCredentialId, projectId, { zone: location.zone });
+}
+
+export function getGKENetworks(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKENetworksResponse> {
+  return getGKEOptions('gkeNetworks', store, cloudCredentialId, projectId, location);
+}
+
+export function getGKESubnetworks(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKESubnetworksResponse> {
+  return getGKEOptions('gkeSubnetworks', store, cloudCredentialId, projectId, location);
+}
+
+export function getGKESharedSubnetworks(store: Store<any>, cloudCredentialId: string, projectId: string, location: {zone?: string, region?: string}): Promise<getGKESharedSubnetworksResponse> {
+  // return getGKEOptions('gkeSharedSubnets', store, cloudCredentialId, projectId, location);
+  // TODO nb remove this test code
+  return Promise.resolve({
```

Comment:
This should be removed before the PR is merged. Our dev GCP setup doesn't have permission to create shared subnets - this data was borrowed from https://github.com/rancher/ui/pull/4790